app/src/main/java/net/programmierecke/radiodroid2/ActivityMain.java

```diff
@@ -678,9 +681,14 @@ public void onFileSelected(FileDialog dialog, File file) {
             Log.i("MAIN", "save to " + file.getParent() + "/" + file.getName());
             RadioDroidApp radioDroidApp = (RadioDroidApp) getApplication();
             FavouriteManager favouriteManager = radioDroidApp.getFavouriteManager();
+            HistoryManager historyManager = radioDroidApp.getHistoryManager();
 
             if (dialog instanceof SaveFileDialog) {
-                favouriteManager.SaveM3U(file.getParent(), file.getName());
+                if (selectedMenuItem == R.id.nav_item_starred) {
+                    favouriteManager.SaveM3U(file.getParent(), file.getName());
+                }else if (selectedMenuItem == R.id.nav_item_history) {
+                    historyManager.SaveM3U(file.getParent(), file.getName());
+                }
```

Comment:
But what about loading the history back?

Reply:
i think history should not be loadable, it is history.

Reply:
you can load it to add it to your favorites

Reply:
But why then export it =) At least in the issue user wanted to backup the history, which in my head means he would also want to restore it.

Reply:
it is not for backup reasons in my opinion, it is to use it with different players.
why would you backup history? if you like a stream you add it to favorites, which you can backup.

Reply:
I agree, I too don't understand it. However in original issue user ask to be able to backup history for whatever reason...

Reply:
i actually believe this is what they meant, but of course i could be wrong...